### PR TITLE
[PS] Clarification on use of MongoSupport vs DefaultPlayMongoRepositorySupport (plus minor tweaks)

### PR DESCRIPTION
**Pull request by beyond-code-github**

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -324,9 +324,7 @@ collection.distinct[String]("name").toFuture()
 
 ### DefaultPlayMongoRepositorySupport
 
-`uk.gov.hmrc.mongo.MongoSpecSupport` should be replaced with `uk.gov.hmrc.mongo.test.DefaultPlayMongoRepositorySupport[T]`.
-
-It expects the repository under test to override `repository`, and provides a `mongoComponent` to create the repository. This will create a database named after the test.
+`uk.gov.hmrc.mongo.MongoSpecSupport` has been replaced with `uk.gov.hmrc.mongo.test.MongoSupport`, however it is reccomended that you migrate to using `uk.gov.hmrc.mongo.test.DefaultPlayMongoRepositorySupport[T]` instead which contains some improvements to the testing experience (**but also changes the way your test data is handled**, see below).
```
Comment (colin-lamed) [inline MIGRATION.md]: typo `recommended`

File: MIGRATION.md [hunk 2]
```diff
@@ -337,9 +335,11 @@ class MyEntityRepositorySpec extends DefaultPlayMongoRepositorySupport[MyEntity]
 }
 ```
 
-`DefaultPlayMongoRepositorySupport` will ensure that the database is cleaned and setup (with indexes and schemas) before each test, and turn on `no table scan` to ensure all queries have an index defined. To refine this behaviour, you may use the composed traits directly.
+`DefaultPlayMongoRepositorySupport[T]` requires you to override `repository` in your tests, and provides a `mongoComponent` to create the repository. This will create a database named after the test.
+
+**NB** - Unlike the old `MongoSpecSupport`, `DefaultPlayMongoRepositorySupport` will ensure that the database is **cleaned of all data** and setup (with indexes and schemas) before each test, and turn on `no table scan` to ensure all queries have an index defined. To refine this behaviour, you may use the composed traits directly.
```
Comment (colin-lamed) [inline MIGRATION.md]: We could be clearer about the extra behaviour coming from additional traits?
e.g.

"**NB** In addition to initialising a mongoComponent (as per `MongoSpecSupport` from reactivemongo-test), `DefaultPlayMonoRepositorySupport` also includes additional traits (akin to `RepositoryPreparation` and `FailOnUnindexedQueries` from reactivemongo-test) which ensure ..."

File: MIGRATION.md [hunk 3]
```diff
@@ -349,6 +349,8 @@ Like MongoSpecSupport, a number of helper functions are provided
 
 The underlying collection can be accessed via `repository.collection`.
 
+You can still use the `MongoSupport` trait directly to more closely replicate the old behaviour.
```
Comment (colin-lamed) [inline MIGRATION.md]: Isn't this covered by the previous comment  "To refine this behaviour, you may use the composed traits directly."? Also seems to imply that previously only the `MongoSupport` trait from reactivemongo-test was used.

Maybe we should move it to it's own paragraph and rephrase it to something like:

"You may prefer to use the underlying traits directly to refine or replicate the old behaviour."

Comment (beyond-code-github) [reply MIGRATION.md]: Yeah good idea, it was more to stress that DefaultPlayMongoRepositorySupport was a new thing that brings together all the behaviours (including other traits), whereas MongoSupport is the direct successor to MongoSpecSupport

**Comment by beyond-code-github** [comment]
Thanks @colin-lamed I have updated based on the feedback

**Review by colin-lamed — APPROVED** [review]
